src/go/rdctl/pkg/config/config.go

```diff
@@ -94,22 +94,23 @@ func GetConnectionInfo(mayBeMissing bool) (*ConnectionInfo, error) {
 		return nil, fmt.Errorf("error parsing config file %q: %w", configPath, err)
 	}
 
-	if connectionSettings.Host == "" {
-		connectionSettings.Host = settings.Host
-		if connectionSettings.Host == "" {
-			connectionSettings.Host = "127.0.0.1"
-		}
+	// CLI options override file settings
+	if connectionSettings.Host != "" {
+		settings.Host = connectionSettings.Host
+	}
+	if settings.Host == "" {
+		settings.Host = "127.0.0.1"
```

Comment:
Would removing the first conditional be acceptable in this case? I think that it would make the code more concise

```suggestion
	settings.Host = connectionSettings.Host
	
	if settings.Host == "" {
		settings.Host = "127.0.0.1"
```

Reply:
@rak-phillip Doesn't this throw away the `host` setting from the config file? It would be either the value of `--host`, or `127.0.0.1`, but the value in the file would always be silently ignored. Why would we want to do that?

Reply:
> but the value in the file would always be silently ignored. Why would we want to do that?

You're right, then it sounds like my suggestion wouldn't be acceptable 🙂

I think that the pattern of `json.Unmarshal(content, &settings)` makes it unclear to me that we are mutating `settings` and that is why I thought it would be acceptable to update `settings` without the need for the conditional. This has more to do with low frequency that I read and write Go more than anything 🙃